crates/circuit/src/circuit_data.rs

```diff
@@ -504,6 +504,25 @@ impl CircuitData {
         Ok(())
     }
 
+    /// Lists all operations along with the qubits involved in those operations.
+    ///
+    /// Returns:
+    ///     list.
+    #[pyo3(signature = ())]
+    pub fn operations_with_qubits(&self, py: Python<'_>) -> PyResult<Py<PyAny>> {
+        let mut ret: Vec<(Option<String>, Option<String>)> = vec![];
+        for (_index, inst) in self.data.iter().enumerate() {
+            let op_str = inst.op.view().name().to_string();
+            for b in self.qargs_interner.get(inst.qubits) {
+                ret.push((
+                    Some(op_str.to_string()),
+                    Some(self.qubits.get(*b).unwrap().to_string()),
```

Comment:
If we were going to use qubit objects like this, I'd have suggested returning them directly instead of stringify them. Then the python user could interact with the bit object directly. However, I don't think we need to work with actual bit objects and can just use indices. If a user wants to map the index to a bit, in python that's just `circuit.qubits[index]`.